### ISSUE TITLE
Plotting predict results

### ISSUE THREAD

**Issue by danieliong** [jxx123/fireTS#5]
Hi Jinyu,

When we are plotting the prediction results when using DirectAutoRegressor, should we be only setting the first `pred_step` elements to be np.nan? 

EDIT: I meant to say should we be plotting the results starting from index `pred_step`. 

I was also wondering why you set the first `pred_step + max(auto_order-1, max(exog_order+exog_delay)-1)` elements to be np.nan. 

Thanks!
Daniel

**Comment by jxx123** [comment]
Hi Daniel, I saw you closed the issue. Have you got the answer?

**Comment by danieliong** [comment]
Hi Jinyu, I wanted to look into it more first but I couldn't delete the issue. When comparing the test target and predictions, should we shift the test target by pred_step? It seems like this makes more sense (for example, the last cell in your example 1. It seems like the the predictions match a shifted version of actual) but wasn't 100% sure.

**Comment by jxx123** [comment]
Hi Daniel, I just wrote up some explanation about the difference between predict and forecast here https://github.com/jxx123/fireTS#faq, and I think it also helps answering your question. 

I just copied the important part here. Given a target time series y(0), y(1), ..., y(9) to predict and the exogenous input time series x(0), x(1), ..., x(9), build a NARX model NARX(RandomForestRegressor(), auto_order=1, exog_order=[1], exog_delay=[0]). The model can be represented by a function y(t + 1) = f(y(t), x(t)) + e(t).
predict(x, y, step=2) outputs a time series that has the same length as original y, and it means the 2-step-ahead prediction at each step, i.e. nan, nan, y_hat(2), y_hat(3), ..., y_hat(9). Note that y_hat(2) is the 2-step-ahead prediction standing at time 0. y_hat(3) is the 2-step-ahead prediction standing at time 1, and so on. Another very important note is that predicted value y_hat(2) = f(y_hat(1), x(1)) = f(f(y(0), x(0)), x(1)). 

The reason that the first 2 values are nan is that there is not enough information to determine the predicted values. For example, to get y_hat(0), you will need y(-2), x(-2), x(-1). To get y_hat(1), you will need y(-1), x(-1), x(0). That is why the first pred_step + max(auto_order-1, max(exog_order+exog_delay)-1) elements have to be np.nan.

For the example 1, the output of predict is already aligned with y. No shift is needed. The reason that it looks like a shifted version is just because the model is weak and all it learned is just relying on the latest y values to do the prediction.
